### PR TITLE
Ready for review

### PR DESCRIPTION
**Pull request by vadimb892**
Capstone project that recreates [rim effect](https://ameye.dev/notes/rendering-outlines/) outline. This project is experiment with [Bevy](https://bevyengine.org/) rust engine for learning purpose. Can be used as lib for adding rim effect outline material extension.

Run demo example where you can change:

- width(power) - press W + mouse scroll;
- time scale - press T + mouse scroll;
- (const width)/(time dependency width) - press R (don`t hold button, otherwise state will change with high frequency).

demo

### CODE REVIEW

**Review by NamesMark — APPROVED** [review]
зображення

Really cool! I am impressed. Cool that you made a demo and provided commands to modify the intensity of the effect. 

Something weird was going on with the command `Outline intensity from time` (`set_mode()`) - sometimes it toggled, sometimes not. But not sure if it's something that depends on you rather than on `bevy`.

Overall, what a great project!

**Review by JekiXD — APPROVED** [review]

**Review by bugsweeper — APPROVED** [review]
Looks good to me

File: examples/demo/asset_loader.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    render::texture::{
+        ImageAddressMode, ImageFilterMode, ImageLoaderSettings, ImageSampler,
+        ImageSamplerDescriptor,
+    },
+};
+
+/// Contain loaded [`Image`] assets
+#[derive(Resource, Debug, Default)]
+pub struct TextureAssets {
+    /// [`TextureSet`] for ground [`StandardMaterial`]
```
Comment (TheBestTvarynka) [inline examples/demo/asset_loader.rs:12]: :+1: 
I like doc comments. Respect!

File: examples/demo/asset_loader.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            )),
+            occlusion: Some(asset_server.load(
+                "textures\\TilesCeramicHerringbone002\\TilesCeramicHerringbone002_AO_1K.jpg",
+            )),
+            depth: Some(asset_server.load(
+                "textures\\TilesCeramicHerringbone002\\TilesCeramicHerringbone002_DISP_1K.jpg",
+            )),
+            ..default()
+        },
+    };
```
Comment (TheBestTvarynka) [inline examples/demo/asset_loader.rs:81]: I do not recommend hardcode paths to files. Use env variables (or any configuration methods) or check in compile time that these files exist

File: src/rim_effect.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+use bevy::{asset::load_internal_asset, prelude::*};
+use bevy::pbr::MaterialExtension;
+use bevy::render::render_resource::ShaderRef;
+
+use super::outline::{Outline, OutlineLabel};
+
+pub const SHADER_HANDLE: Handle::<Shader> = 
+    Handle::<Shader>::weak_from_u128(42572727525244273574);
```
Comment (TheBestTvarynka) [inline src/rim_effect.rs:8]: it's better to add a comment about what this number mean

**Review by TheBestTvarynka — APPROVED** [review]

File: src/rim_effect.rs (hunk 3 above)
Comment (bugsweeper) [reply src/rim_effect.rs:8]: It's random value, handle should be unique

**Review by ukrustacean — APPROVED** [review]
This is what I call clean code =)

**Pull request merged**